jax/_src/scipy/stats/vonmises.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import scipy.stats as osp_stats
+from jax import lax
+from jax._src.lax.lax import _const as _lax_const
+from jax._src.numpy import lax_numpy as jnp
+from jax._src.numpy.util import _wraps, _promote_args_inexact
+from jax._src.typing import Array, ArrayLike
+
+def _cosm1(x: ArrayLike) -> Array:
+  minus_one = _lax_const(x, -1)
+  return lax.add(lax.cos(x), minus_one)
+
+def _i0e(x: ArrayLike) -> Array:
+  return lax.exp(-lax.abs(x)) * jnp.i0(x)
+
+def _pdf(x: ArrayLike, kappa: ArrayLike) -> Array:
+  pi = _lax_const(kappa, jnp.pi)
+  return lax.exp(kappa * _cosm1(x)) / (2 * pi * _i0e(kappa))
```

Comment:
Hi - it looks like there are some errors when `JAX_ENABLE_X64=1` (you should be able to see the "Tests with internal numpy/scipy" failures in the CI list) One of these binary operations is happening between two values that have mismatched types. We check for this in the test suite in order to support `jax_numpy_dtype_promotion='strict'` without false positives. I think the issue is that `lax_const` returns a numpy scalar, and so `2 * pi` is a numpy operation that promotes to float64. If you avoid the const and just write `2 * jnp.pi * lax.bessel_i0e(kappa)` I think it should fix the issue.